### PR TITLE
AzureRT PR - update vm quick-create, use NRP autoRest-client, & other fixes

### PR DESCRIPTION
**Pull request by huangpf**

### CODE REVIEW

**Comment by huangpf** [comment]
@yugangw-msft @amarzavery Could you please help take a look?

File: lib/commands/arm/vm/vmClient._js [hunk 1]
```diff
@@ -201,13 +201,15 @@ __.extend(VMClient.prototype, {
     params.imageUrn = imageUrn;
     // hardwareProfile
     // For quick create requirement is to have 'Standard_D1' as default
-    params.vmSize = 'Standard_D1';
+    params.vmSize = options.vmSize ? options.vmSize : 'Standard_D1';
```
Comment (yugangw-msft) [inline lib/commands/arm/vm/vmClient._js]: nit: you can simplify with (options.vmSize || 'Standard_D1')

File: lib/commands/arm/vm/vmClient._js [hunk 2]
```diff
@@ -201,13 +201,15 @@ __.extend(VMClient.prototype, {
...
     // osProfile
     params.computerName = params.vmName;
     params.adminUsername = adminUsername;
     params.adminPassword = adminPassword;
+    params.sshPublickeyFile = options.sshPublickeyFile;
     params.osType = osType;
     // storageProfile - storage accountweb
+    params.storageAccountName = ('cli' +  hash + utils.getHash(adminUsername)).substring(0, 24);
```
Comment (yugangw-msft) [inline lib/commands/arm/vm/vmClient._js]: i don't have the domain knowledge, but generally speaking, use hard-coded 0 and 24 to retrieve substring looks unprotected on error cases or new cases.

File: scripts/unit.js [hunk 3]
```diff
@@ -38,7 +38,7 @@ function buildFileList(testFiles, testList, root) {
   var includeMark = 'include:';
   for (var i = 0; i < files.length ; i++) {
     if (files[i].indexOf(includeMark) === 0) {
-      var fileToInclude = files[i].substring(includeMark.length).replace('\r', '');
+      var fileToInclude = files[i].substring(includeMark.length).replace(/\s+$/g, '');
```
Comment (yugangw-msft) [inline scripts/unit.js]: please back out the change, using "\r' " is the fix, which is consistent with the other code

**Comment by yugangw-msft** [comment]
@romangromov, please take a look, this touches network code. I think it is due to expected interface change

File: lib/commands/arm/vm/vmClient._js (hunk 2 above)
Comment (huangpf) [inline lib/commands/arm/vm/vmClient._js]: Node.js' substring function will extract 'up to' the ```end``` position, but not including. So I think it should be fine here.

http://www.w3schools.com/jsref/jsref_substring.asp

```
Parameter	Description
start	Required. The position where to start the extraction. First character is at index 0
end	Optional. The position (up to, but not including) where to end the extraction. If omitted, it extracts the rest of the string
```

**Comment by romangromov** [comment]
@yugangw-msft reviewed, this PR do not affect any `azure network` command, changes related to Networking SDK from autorest looks good to me

**Pull request merged**